### PR TITLE
[datafactory] Regenerate datafactory to support new linked service types

### LINKED ISSUE #2872: Azure Data Factory Extension - Linked Service Creation - not reading JSON file
## Functionality Feedback

When I am trying to run a command to create a linked service where the contents of the properties are stored in JSON file, it is not reading them. It is sending a request to the API with a request body which looks like this - 

```
{"properties": {"type": "LinkedService"}}
```

Here is the debug log for the request:
```
azure.core.pipeline.policies._universal : Request URL: 'https://management.azure.com/subscriptions/(redacted-subscription)/resourceGroups/(redacted-rg)/providers/Microsoft.DataFactory/factories/(readacted-adf-name)/linkedservices/HyasDataPrep2?api-version=2018-06-01'
azure.core.pipeline.policies._universal : Request method: 'PUT'
azure.core.pipeline.policies._universal : Request headers:
azure.core.pipeline.policies._universal :     'Content-Type': 'application/json'
azure.core.pipeline.policies._universal :     'Accept': 'application/json'
azure.core.pipeline.policies._universal :     'Content-Length': '41'
azure.core.pipeline.policies._universal :     'x-ms-client-request-id': '28f918d4-5135-11eb-b3a4-0242ac110002'
azure.core.pipeline.policies._universal :     'CommandName': 'datafactory linked-service create'
azure.core.pipeline.policies._universal :     'ParameterSetName': '--factory-name --name --properties --resource-group --debug'
azure.core.pipeline.policies._universal :     'User-Agent': 'AZURECLI/2.14.0 (DEB) azsdk-python-datafactorymanagementclient/unknown Python/3.6.10 (Linux-4.19.128-microsoft-standard-x86_64-with-debian-10.6)'
azure.core.pipeline.policies._universal :     'Authorization': '*****'
azure.core.pipeline.policies._universal : Request body:
azure.core.pipeline.policies._universal : {"properties": {"type": "LinkedService"}}
```
The `az-cli` command is -

```
az datafactory linked-service create --factory-name "redacted-factory-name" --name "linked-service" --properties @./"linked-services/somelinkedservice.json" --resource-group "redacted-rg" --debug
```

## Design Feedback

The JSON payload for the `properties` is very different what is expressed in ADF Authoring tool, when you look at the JSON declaration on the service. 

For example: 

- JSON Payload for a Storage account in Portal/Azure Data Factory Authoring tool

``` json
{
    "name": "AzureDataLakeStorage1",
    "type": "Microsoft.DataFactory/factories/linkedservices",
    "properties": {
        "annotations": [],
        "type": "AzureBlobFS",
        "typeProperties": {
            "url": "https://astoreageaccountincloud.dfs.core.windows.net",
            "encryptedCredential": "superdupersecretkey"
        }
    }
}
```

- JSON Payload required by `az datafactory linked-service create` command

```
{
 "type": "AzureStorage",
  "typeProperties": {
        "connectionString" :{ 
             "type": "SecureString",
             "value":"DefaultEndpointsProtocol=https;AccountName=examplestorageaccount;AccountKey=<storage key>"
          }
     }
}
```

Unlike the `az datafactory pipeline create` command, which seems to accept the same JSON as it is in the pipeline code in the Authoring tool.

I think, for ease of developer experience and consistency, the JSON payload should be same as the one in the portal/authoring tool. . The one in `az datafactory pipeline create`  is the better model. I am hoping the API is consistent as well.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 13386c48-1ac8-9452-fcb3-263829e9982c
* Version Independent ID: 1060fc99-cdb7-9c06-2374-f3127fa5e3f5
* Content: [az datafactory linked-service](https://docs.microsoft.com/en-us/cli/azure/ext/datafactory/datafactory/linked-service?view=azure-cli-latest#ext_datafactory_az_datafactory_linked_service_create)
* Content Source: [latest/docs-ref-autogen/ext/datafactory/datafactory/linked-service.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/latest/docs-ref-autogen/ext/datafactory/datafactory/linked-service.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### LINKED ISSUE #3223: Not able to deploy AzureDatabricksDeltaLake for ADF linked service by az datafactory linked-service create
- If the issue is to do with Azure CLI 2.0 in-particular, create an issue here at [Azure/azure-cli](https://github.com/Azure/azure-cli/issues)

### Extension name (the extension in question)
az datafactory linked-service create

### Description of issue (in as much detail as possible)
AzureDatabricksDeltaLake has been newly added as a liked service resource in ADF and it is currently not supported with Azure CLI deployment.
Can you please consider if AzureDatabricksDeltaLake can be deployed with Azure CLI?

-----

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve #3223
Resolve #2872

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by qiaozha** [comment]
@fengzhou-msft There're breaking changes between the previous generated code and now, such as `az datafactory factory create` => `az datafactory create`. I think you need to update the previous manual override part and the test part as well

File: src/datafactory/azext_datafactory/generated/custom.py [hunk 1]
```diff
@@ -553,8 +581,10 @@ def datafactory_pipeline_update(instance,
         instance.annotations = annotations
     if run_dimensions is not None:
         instance.run_dimensions = run_dimensions
+    if duration is not None:
+        instance.elapsed_time_metric.duration = duration
     if folder_name is not None:
-        instance.name_folder_name = folder_name
+        instance.folder.name = folder_name
```
Comment (qiaozha) [inline src/datafactory/azext_datafactory/generated/custom.py:587]: Is this folder_name  flattened from folder.name ? I wonder if this instance path is correct ?

File: src/datafactory/azext_datafactory/manual/custom.py [hunk 2]
```diff
@@ -11,30 +11,45 @@
...
-def datafactory_factory_create(client,
-                               resource_group_name,
-                               factory_name,
-                               if_match=None,
-                               location=None,
-                               tags=None,
-                               factory_vsts_configuration=None,
-                               factory_git_hub_configuration=None,
-                               global_parameters=None):
+def datafactory_create(client,
```
Comment (qiaozha) [inline src/datafactory/azext_datafactory/manual/custom.py:14]: wondering why do we need to overide this? looks the same as the generated code ?

**Comment by qiaozha** [comment]
@changlong-liu Could you also take a look at this PR ? see if all these changes related with generated test or examples are expected ? Thanks

**Comment by changlong-liu** [comment]
Should be fine. The test breaking change comes from some early requirements like https://github.com/Azure/autorest.az/issues/600

Looks much manual efforts were spent on fixing test breaking changes. We may need to be more cautious on the test interface changes later.

File: src/datafactory/azext_datafactory/manual/custom.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/datafactory/azext_datafactory/manual/custom.py:14]: In generated code, identity is empty: `factory['identity'] = {}`. Portal behavior is enabling system assigned identity by default, the change aligns the behavior with Portal.

File: src/datafactory/azext_datafactory/generated/custom.py (hunk 1 above)
Comment (fengzhou-msft) [reply src/datafactory/azext_datafactory/generated/custom.py:587]: Yes. I think it's correct.

File: src/datafactory/azext_datafactory/manual/custom.py (hunk 2 above)
Comment (qiaozha) [reply src/datafactory/azext_datafactory/manual/custom.py:14]: Maybe can try to set default value for identity ?

Comment (qiaozha) [reply src/datafactory/azext_datafactory/manual/custom.py:14]: and hide this parameter with directive

Comment (fengzhou-msft) [reply src/datafactory/azext_datafactory/manual/custom.py:14]: The parameter is already hidden. The default value is not a str, but an object, is it able to do so?

File: src/datafactory/azext_datafactory/generated/action.py [hunk 3]
```diff
@@ -101,6 +109,9 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
             v = properties[k]
             if kl == 'name':
                 d['name'] = v[0]
+            else:
+                raise CLIError('Unsupported Key {} is provided for parameter folder. All possible keys are: name'.
```
Comment (qwordy) [inline src/datafactory/azext_datafactory/generated/action.py:113]: `CLIError` is deprecated

File: src/datafactory/setup.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 from setuptools import setup, find_packages
 
 # HISTORY.rst entry.
-VERSION = '0.2.0'
+VERSION = '0.1.0'
```
Comment (qwordy) [inline src/datafactory/setup.py:13]: Can version decrease?

File: src/datafactory/azext_datafactory/generated/action.py (hunk 3 above)
Comment (fengzhou-msft) [reply src/datafactory/azext_datafactory/generated/action.py:113]: Generated code in extensions still uses `CLIError` to be compatible with older version of CLI.

File: src/datafactory/setup.py (hunk 4 above)
Comment (fengzhou-msft) [reply src/datafactory/setup.py:13]: Caused by codegen change. The `VERSION` here will now be overwritten by `from azext_datafactory.manual.version import VERSION` and that is changed to `0.3.0` manually.

**Review by qwordy — APPROVED** [review]

**Pull request merged**